File: music-api/src/main/java/com/music/sale/web/category/mapper/CategoryQueryWebMapper.java
package com.music.sale.web.category.mapper;

import com.music.sale.domain.category.Category;
import com.music.sale.web.category.response.CategoryQueryResponse;
import java.util.List;
import org.springframework.stereotype.Component;

@Component
public class CategoryQueryWebMapper {

  public CategoryQueryResponse toResponse(Category category) {
    if (category == null) {
      return null;
    }

    List<CategoryQueryResponse> childrenResponses =
        category.getChildren().stream()
            .map(this::toResponse)
            .collect(java.util.stream.Collectors.toList());

    return new CategoryQueryResponse(
        category.getId(),
        category.getName(),
        category.getName(),
        null,
        category.getType().name(),
        category.getParent() != null ? category.getParent().getId() : null,
        childrenResponses);
  }
}

File: music-api/src/main/kotlin/com/music/sale/web/category/mapper/CategoryQueryWebMapper.kt
package com.music.sale.web.category.mapper

import com.music.sale.domain.category.Category
import com.music.sale.web.category.response.CategoryQueryResponse
import org.springframework.stereotype.Component

@Component
class CategoryQueryWebMapper {
    fun toResponse(category: Category): CategoryQueryResponse {
        return CategoryQueryResponse(
            id = category.id,
            name = category.name,
            type = category.type,
            parent = category.parent,
            path = category.path,
            depth = category.depth,
            isActive = category.isActive,
        )
    }
}
